### PR TITLE
Clean reconfDocker.sh

### PR DESCRIPTION
**Pull request by jjmaestro**
- remove dirty sleep for a wait loop
- break long commands in multiple lines
- clarify error logging, giving the full filename in the wait loop
- remove unneeded comments

Related to #14912

### CODE REVIEW

**Comment by dalanlan** [comment]
You may need to sign the cla:)

**Comment by jjmaestro** [comment]
@dalanlan yup, but I need to clear it with my employer, so it might take a little while. I'll try to get it ASAP!

File: cluster/ubuntu/reconfDocker.sh [hunk 1]
```diff
@@ -46,15 +46,26 @@ function config_etcd {
...
 
 function restart_docker {
-  #wait some secs for /run/flannel/subnet.env ready
-  sleep 15
   sudo ip link set dev docker0 down
   sudo brctl delbr docker0
 
+  attempt=0
+  while [ ! -f /run/flannel/subnet.env ]; do
+    if (( attempt > 600 )); then
```
Comment (brendandburns) [inline cluster/ubuntu/reconfDocker.sh]: Perhaps echo "Waiting for up to 30 minutes for flannel..."

**Comment by brendandburns** [comment]
Thanks for the PR!  basically LGTM, we do need the CLA.

Thanks!
--brendan

File: cluster/ubuntu/reconfDocker.sh (hunk 1 above)
Comment (jjmaestro) [inline cluster/ubuntu/reconfDocker.sh]: :+1: sure, I'll add those

**Comment by jjmaestro** [comment]
@brendandburns cool, thanks :) I've sent an email to the legal PoC in my company, will see what they say about the CLA!

**Comment by dalanlan** [comment]
This patch could introduce a potential issue. 
Assuming we start a cluster from a dirty env which had an old `/run/flannel/subnet.env` ( yeah we should clear that one in `kube-down` process, which i'm working on currently), the detection could lead us to the wrong direction. 
We wait enough time (15s) to get rid of that earlier, making no guarantee yet more safe.

**Comment by dalanlan** [comment]
Please merge this after #15135

**Comment by jjmaestro** [comment]
@dalanlan IMHO, you yourself said it, `kube-down` should clear state :) having an arbitrary wait in there will not be of any help. I'll make sure to merge #15135, I'm still waiting for clearance to sign the CLA, so it's OK for me to wait a bit more.

File: cluster/ubuntu/reconfDocker.sh [hunk 2]
```diff
@@ -32,13 +31,14 @@ function config_etcd {
     if [[ "$?" == 0 ]]; then
       break
     else
-    	# enough timeout??
       if (( attempt > 600 )); then
-        echo "timeout for waiting network config" > ~/kube/err.log
+        echo "timeout waiting for /coreos.com/network/config" >> ~/kube/err.log
```
Comment (dalanlan) [inline cluster/ubuntu/reconfDocker.sh]: good catch:)

**Comment by dalanlan** [comment]
@jjmaestro Are you still working around this, if i may ask:)

**Comment by jjmaestro** [comment]
@dalanlan I talked to legal and, unfortunately, I'm not clear to work on kubernetes :( so, I'll close the pull requests. Sorry about that, I should have cleared it beforehand, but I wasn't really anticipating this outcome.

**Comment by dalanlan** [comment]
@jjmaestro Yet it was a good job. I would send anther pr to carry this if you don't mind. Thanks all the same.

**Pull request closed without merging**